contrib/platform/src/com/sun/jna/platform/WindowUtils.java

```diff
@@ -1285,17 +1285,30 @@ public String getProcessFilePath(final HWND hwnd) {
 
             final HANDLE process = Kernel32.INSTANCE.OpenProcess(WinNT.PROCESS_QUERY_INFORMATION | WinNT.PROCESS_VM_READ,
                                                                  false, pid.getValue());
-            if (process == null
-                && Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_ACCESS_DENIED) {
-                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            if (process == null) {
+                if(Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_ACCESS_DENIED) {
+                    throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+                } else {
+                    // Ignore windows, that can't be accessed
+                    return "";
+                }
             }
-            final int length = Psapi.INSTANCE.GetModuleFileNameExW(process,
-                                                                   null, filePath, filePath.length);
-            if (length == 0
-                && Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_INVALID_HANDLE) {
-                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+
+            try {
+                final int length = Psapi.INSTANCE.GetModuleFileNameExW(process,
+                                                                       null, filePath, filePath.length);
+                if (length == 0) {
+                    if(Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_INVALID_HANDLE) {
+                        throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+                    } else {
+                        // ignore invalid handles
+                        return "";
```

Comment:
Same here. The exception should be thrown on any other error, the previous code was correct IMO.

Reply:
You mean, that the function(s) should not return the empty string in case of an error? Yeah, I thought about "fixing" that, but I decided against it, as it basically became API by convention.

There are other thinks I would change looking at it - seeing a `HWND` in a common class is suspicious.